### PR TITLE
karma-webpack-babel-vuejs test stack

### PR DESCRIPTION
**Pull request by misstick**
- [x] add a conf that work on local env,
 - [x] add `mocha-reporter` to show test into log,
 - [x] make it work on Travis.

see https://github.com/cozy-labs/konnectors/pull/587 for `package.json` changes
see https://vue-loader.vuejs.org/en/workflow/testing.html

### CODE REVIEW

File: client/webpack.config.js [hunk 1]
```diff
@@ -1,16 +1,17 @@
 "use strict";
```
Comment (m4dz) [inline client/webpack.config.js]: If you clean this file, can you also fix this with a single quote and no semi-colon? thanks.

File: client/webpack.test.config.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+"use strict"
```
Comment (m4dz) [inline client/webpack.test.config.js]: `'use strict'` must be singlequoted

File: client/webpack.test.config.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        postcss: [
+            require('autoprefixer')(['last 2 versions']),
+            require('css-mqpacker')
+        ]
+    },
+
+    stylus: {
+        use: [require('cozy-ui/lib/stylus')()]
+    }
+};
```
Comment (m4dz) [inline client/webpack.test.config.js]: Maybe I'm foolish, but I don't understand where this file is used?

File: package.json [hunk 4]
```diff
@@ -41,8 +41,11 @@
   },
   "devDependencies": {
     "babel-cli": "6.11.4",
-    "babel-preset-es2015": "6.6.0",
+    "babel-core": "^6.18.2",
```
Comment (m4dz) [inline package.json]: Please, don't use `^` version to ensure environments are stucked to specific versions. Can you remove them from the package.json file?

File: package.json [hunk 5]
```diff
@@ -51,26 +54,29 @@
...
-    "jsdom-global": "2.1.0",
-    "mockery": "2.0.0"
+    "webpack": "1.13.2"
   },
   "scripts": {
     "build": "sh scripts/build.sh",
     "test": "sh scripts/lint.sh && npm-run-all test:*",
     "test:server": "sh scripts/tests.sh",
-    "test:client": "mocha --require babel-register client/test/**/*.js",
+    "test:client": "sh scripts/tests_client.sh",
```
Comment (m4dz) [inline package.json]: Please, don't add new tests in external scripts, but rely on _npm scripts_ and its `pre`/`post` scripts. It do let it be more maintainable / readable (see comments in the script file). Thanks.

File: scripts/tests_client.sh [hunk 6]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+echo "Install clients dependencies..."
```
Comment (m4dz) [inline scripts/tests_client.sh]: This task can be done in a `pretest:client` task (see `prewatch:client`)

File: scripts/tests_client.sh [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+cd ./client
+rm -rf node_modules/
```
Comment (m4dz) [inline scripts/tests_client.sh]: `rm` dir is useless as this dir isn't versionned. You can safely avoid it.

**Comment by misstick** [comment]
@m4dz it seems that your proposal doesnt work with travis; do you have any idea why using an external file works an not this one?

**Comment by m4dz** [comment]
My proposal perfectly works, and Karma is propoerly running. But you can't expect having Firefox running on Travis as it's a non-xwindow aware environment. You should take a look at https://karma-runner.github.io/1.0/plus/travis.html.

**Comment by misstick** [comment]
I will look at this tomorrow; I've just looked at "succeed build" and there was other errors but that didnt have failed build; sorry for not having seen this before.

**Comment by m4dz** [comment]
Thanks!

**Pull request merged**